app/models/bundle_context.rb

```diff
@@ -21,4 +26,82 @@ class BundleContext < ApplicationRecord
     "smpl_style" => 2
   }
 
+  def staging_dir
+    '/dor/assembly'
+  end
+
+  def normalize_bundle_dir
+    self[:bundle_dir].chomp("/") if bundle_dir
+  end
+
+  def progress_log_file
+    Tempfile.new.path(id) #FIXME: (#78)
+  end
+
+  def content_exclusion #FIXME: Delete everywhere in code (#227)
+    nil
+  end
+
+  def file_attr
+    nil # FIXME can get rid of this (#228)
+  end
+
+  def validate_files?
+    false #FIXME delete everwhere in code (#230)
+  end
+
+  def content_tag_override? #TODO: find where this is used as a conditional and delete code that won't be executed (#231)
+    true
+  end
+
+  def smpl_manifest
+    'smpl_manifest.csv'
+  end
+
+  def manifest
+    'manifest.csv'
+  end
+
+  def path_in_bundle(rel_path)
+    File.join(bundle_dir, rel_path)
+  end
+
+  # TODO: BundleContext is not really a logical home for this util method
```

Comment:
hrmm - not sure I agree, as reading the file from the user supplied location is needed to flesh out the bundle context data ...

Also - why is this a class method?  should it be a private instance method directly operating on path_in_bundle(manifest) ?   I think this is the only place where we read a csv file, unless @atz envisions writing out discovery report data as a csv (which could be a good idea, if folks work with the old discovery report that way - as it was designed)

Reply:
ah, i'd left this comment a while back (after splitting out `BundleContext` from `Bundle`, but before sarav moved it to its own file).  at that time, `Bundle` also used `import_csv`.  but i just looked, and it doesn't use it any longer.  so yeah, now it could just be a private instance method in this class, which is now a logical home for it.

i think that also means we can get rid of this delegation, and the `require 'csv'` at the top of `bundle.rb`:
https://github.com/sul-dlss/pre-assembly/blob/master/app/lib/pre_assembly/bundle.rb#L3
https://github.com/sul-dlss/pre-assembly/blob/master/app/lib/pre_assembly/bundle.rb#L41-L43